src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs

```diff
@@ -236,7 +236,8 @@ private IActionResult ToSaveOutcomeResult(SaveOutcome saveOutcome)
                 case SaveOutcomeType.Updated:
                     return FhirResult.Create(saveOutcome.RawResourceElement, HttpStatusCode.OK)
                         .SetETagHeader()
-                        .SetLastModifiedHeader();
+                        .SetLastModifiedHeader()
+                        .SetLocationHeader(_urlResolver);
```

Comment:
https://www.hl7.org/fhir/http.html#update
>If the interaction is successful, the server SHALL return either a 200 OK HTTP status code if the resource was updated, or a 201 Created status code if the resource was created (or brought back to life/re-created)

I disagree with this changes, for the case you looking for we should get `Created` OutcomeType not `Updated`, this is just hiding problem, rather than address it.

Reply:
I like the idea where for conditional updates it also provides the location in the response as the id wasn't known. It is not in the spec, but it seemed useful. I'll put it in the icebox for now.